src/common/components/CustomIcon/SchoolIcon.tsx

```diff
@@ -0,0 +1,1099 @@
+import { CustomIcon, type CustomIconProps } from "./CustomIcon";
+
+export interface SchoolIconProps extends CustomIconProps {
+  school: "NTU" | "SMU" | "NUS";
+}
+
+export const SchoolIcon: React.FC<SchoolIconProps> = ({ school, ...props }) => {

```

Comment:
We can avoid adding `React.FC` for the return type! It is no longer recommended to use it and you can check [React official guide here](https://react.dev/learn/typescript#typescript-with-react-components)! 

[Facebook](https://github.com/facebook/create-react-app/pull/8177) removed it in their template as well

Reply:
Amended!